opentelemetry-sdk/src/trace/mod.rs

```diff
@@ -36,3 +36,62 @@ pub use sampler::{JaegerRemoteSampler, JaegerRemoteSamplerBuilder};
 
 #[cfg(test)]
 mod runtime_tests;
+
+#[cfg(all(test, feature = "testing"))]
+mod tests {
+    use super::*;
+    use crate::testing::trace::InMemorySpanExporterBuilder;
+    use opentelemetry::{
+        trace::{Span, Tracer, TracerProvider as _},
+        KeyValue,
+    };
+
+    #[test]
+    fn tracing_in_span_test() {
+        // Arrange
+        let exporter = InMemorySpanExporterBuilder::new().build();
+        let provider = TracerProvider::builder()
+            .with_span_processor(SimpleSpanProcessor::new(Box::new(exporter.clone())))
+            .build();
+
+        // Act
+        let tracer = provider.tracer("test_tracer");
+        tracer.in_span("span_name", |_cx| {});
+
+        provider.force_flush();
+
+        // Assert
+        let exported_spans = exporter
+            .get_finished_spans()
+            .expect("Spans are expected to be exported.");
+        assert_eq!(exported_spans.len(), 1);
+        let span = &exported_spans[0];
+        assert_eq!(span.name, "span_name");
+        assert_eq!(span.instrumentation_lib.name, "test_tracer");
+    }
+
+    #[test]
+    fn tracing_tracer_start_test() {
+        // Arrange
+        let exporter = InMemorySpanExporterBuilder::new().build();
+        let provider = TracerProvider::builder()
+            .with_span_processor(SimpleSpanProcessor::new(Box::new(exporter.clone())))
+            .build();
+
+        // Act
+        let tracer = provider.tracer("test_tracer");
+        let mut span = tracer.start("span_name");
+        span.set_attribute(KeyValue::new("key1", "value1"));
+        drop(span);
+        provider.force_flush();
```

Comment:
This part 
```
let exporter = InMemorySpanExporterBuilder::new().build();
        let provider = TracerProvider::builder()
            .with_span_processor(SimpleSpanProcessor::new(Box::new(exporter.clone())))
            .build();
```
seems to be a common pattern tests will use can we extract it as `setup` function?

Reply:
I'll defer it now. There are quite a number of tests coming, and once they all land, I can extract duplicates to helper method. I expect even setup would be different for many, so might need more customization than just a simple setup() function.